// @aws-sdk/client-cognito-identity/dist-cjs/runtimeConfig.shared.js
const { AwsSdkSigV4Signer } = require("@aws-sdk/core/httpAuthSchemes");
const { AwsJson1_1Protocol } = require("@aws-sdk/core/protocols");
const { NoAuthSigner } = require("@smithy/core");
const { NoOpLogger } = require("@smithy/core/client");
const { parseUrl } = require("@smithy/core/protocols");
const { fromBase64, fromUtf8, toBase64, toUtf8 } = require("@smithy/core/serde");
const { defaultCognitoIdentityHttpAuthSchemeProvider } = require("./auth/httpAuthSchemeProvider");
const { defaultEndpointResolver } = require("./endpoint/endpointResolver");
const { errorTypeRegistries } = require("./schemas/schemas_0");
exports.getRuntimeConfig = (config) => {
    return {
        apiVersion: "2014-06-30",
        base64Decoder: config?.base64Decoder ?? fromBase64,
        base64Encoder: config?.base64Encoder ?? toBase64,
        disableHostPrefix: config?.disableHostPrefix ?? false,
        endpointProvider: config?.endpointProvider ?? defaultEndpointResolver,
        extensions: config?.extensions ?? [],
        httpAuthSchemeProvider: config?.httpAuthSchemeProvider ?? defaultCognitoIdentityHttpAuthSchemeProvider,
        httpAuthSchemes: config?.httpAuthSchemes ?? [
            {
                schemeId: "aws.auth#sigv4",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4"),
                signer: new AwsSdkSigV4Signer(),
            },
            {
                schemeId: "smithy.api#noAuth",
                identityProvider: (ipc) => ipc.getIdentityProvider("smithy.api#noAuth") || (async () => ({})),
                signer: new NoAuthSigner(),
            },
        ],
        logger: config?.logger ?? new NoOpLogger(),
        protocol: config?.protocol ?? AwsJson1_1Protocol,
        protocolSettings: config?.protocolSettings ?? {
            defaultNamespace: "com.amazonaws.cognitoidentity",
            errorTypeRegistries,
            xmlNamespace: "http://cognito-identity.amazonaws.com/doc/2014-06-30/",
            version: "2014-06-30",
            serviceTarget: "AWSCognitoIdentityService",
        },
        serviceId: config?.serviceId ?? "Cognito Identity",
        urlParser: config?.urlParser ?? parseUrl,
        utf8Decoder: config?.utf8Decoder ?? fromUtf8,
        utf8Encoder: config?.utf8Encoder ?? toUtf8,
    };
};
